src/lib/Simplify.hs

```diff
@@ -481,6 +517,16 @@ simplifyAtom atom = confuseGHC >>= \_ -> case atom of
   DepPairRef _ _ _ -> error "Should only occur in Imp lowering"
   ProjectElt idxs v -> getProjection (toList idxs) <$> simplifyVar v
 
+-- We use a different name because the implementations might need to diverge.
```

Comment:
This seems like a bad idea to me, especially that you've already missed a few points where types might appear (e.g. data con params), so it's already misleading. So perhaps just don't do this until we really need to, and then we'll do it properly?

Reply:
sgtm